components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/api/auth/ApiAuthnHandler.java

```diff
@@ -0,0 +1,230 @@
+                authenticatorData.getIdp()));
+        authenticator.setAuthenticator(authenticatorData.getDisplayName());
+        authenticator.setIdp(authenticatorData.getIdp());
+        AuthenticatorMetadata metadata = buildAuthenticatorMetadata(authenticatorData);
+        authenticator.setMetadata(metadata);
+        authenticator.setRequiredParams(authenticatorData.getRequiredParams());
+        return authenticator;
+    }
+
+    private List<Message> buildMessages(AuthServiceResponse authServiceResponse) {
+
+        List<Message> messages = new ArrayList<>();
+        return messages;
+    }
+
+    private AuthenticatorMetadata buildAuthenticatorMetadata(AuthenticatorData authenticatorData) {
+
+        AuthenticatorMetadata authenticatorMetadata = new AuthenticatorMetadata();
+        authenticatorMetadata.setI18nKey(authenticatorData.getI18nKey());
+        if (isAdditionalAuthenticatorDataAvailable(authenticatorData)) {
+            authenticatorMetadata.setPromptType(authenticatorData.getPromptType());
+            List<Param> params = new ArrayList<>();
+            for (AuthenticatorParamMetadata paramMetadata : authenticatorData.getAuthParams()) {
+                Param param = buildAuthenticatorParam(paramMetadata);
+                params.add(param);
+            }
+            authenticatorMetadata.setParams(params);
+            authenticatorMetadata.setAdditionalData(getAdditionalData(authenticatorData.getAdditionalData()));
+        }
+        return authenticatorMetadata;
+    }
+
+    private boolean isAdditionalAuthenticatorDataAvailable(AuthenticatorData authenticatorData) {
+
+        // We can assume if required params are empty that other additional data is also empty.
+        return !authenticatorData.getRequiredParams().isEmpty();
+    }
+
+    private Map<String, String> getAdditionalData(AdditionalData additionalData) {
+
+        Map<String, String> additionalDataMap = new HashMap<>(additionalData.getAdditionalAuthenticationParams());
+        if (StringUtils.isNotBlank(additionalData.getRedirectUrl())) {
+            additionalDataMap.put(ADDITIONAL_DATA_REDIRECT_URL, additionalData.getRedirectUrl());
+        }
+        return additionalDataMap;
+    }
+
+    private Param buildAuthenticatorParam(AuthenticatorParamMetadata paramMetadata) {
+
+        Param param = new Param();
+        param.setParam(paramMetadata.getName());
+        param.setType(paramMetadata.getType());
+        param.setConfidential(paramMetadata.isConfidential());
+        param.setOrder(paramMetadata.getParamOrder());
+        param.setI18nKey(paramMetadata.getI18nKey());
+
+        return param;
+    }
+
+    private List<Link> buildLinks() throws AuthServiceException {
+
+        List<Link> links = new ArrayList<>();
+        Link authnEpLink = new Link();
+        authnEpLink.setName("authentication");
+        String endpoint = "/oauth2/authn";
```

Comment:
Let's define these as constants in OAuthConstants.

Reply:
fixed. Moved to class level constants as currently these are not used in any other place